src/main/java/su/litvak/chromecast/api/v2/ChromeCasts.java

```diff
@@ -26,56 +26,57 @@
 /**
  * Utility class that discovers ChromeCast devices and holds references to all of them.
  */
-public class ChromeCasts extends ArrayList<ChromeCast> implements ServiceListener {
+public class ChromeCasts {
     private final static ChromeCasts INSTANCE = new ChromeCasts();
+    private final MyServiceListener listener = new MyServiceListener();
 
     private JmDNS mDNS;
 
-    private List<ChromeCastsListener> listeners = new ArrayList<ChromeCastsListener>();
-
+    private final List<ChromeCastsListener> listeners = new ArrayList<ChromeCastsListener>();
+    private final List<ChromeCast> chromeCasts = new ArrayList<ChromeCast>();
+    
     private ChromeCasts() {
     }
-
-    private void _startDiscovery(InetAddress addr) throws IOException {
-        if (mDNS == null) {
-            if(addr != null) {
-                mDNS = JmDNS.create(addr);
-            } else {
-                mDNS = JmDNS.create();
-            }            
-            mDNS.addServiceListener(ChromeCast.SERVICE_TYPE, this);
-        }
-    }
-
-    private void _stopDiscovery() throws IOException {
-        if (mDNS != null) {
-            mDNS.close();
-            mDNS = null;
+    
+    /** Returns a copy of the currently seen chrome casts.
+     * @return a copy of the currently seen chromecast devices.
+     */
+    public static List<ChromeCast> get() {
+        synchronized (INSTANCE.chromeCasts) {
```

Comment:
Ok, let's stick with a synchronized list per your new changes. Thanks for considering this.